django_webapps/scripts/db_population/populate_teachers.py

```diff
@@ -38,25 +78,147 @@ def load_teachers(path_to_csv_file: str) -> list[list[str]]:
 
 
 class TeacherPopulator:
+    def __init__(self, teachers: list[list[str]]) -> None:
+        self.teachers = teachers[1:]
+        self.header = teachers[0]
+        self._current_teacher = None
+
     def run(self) -> None:
-        pass
+        for teacher in tqdm(self.teachers, desc="Going through the tachers..."):
+            tid = teacher[COLUMN_TO_ID["tid"]]
+            logger.info("======================================================= ")
+            logger.info(f"Parsing teacher with tid {tid}")
+            self._current_teacher = teacher
+            self._create_teacher()
 
+    @transaction.atomic
     def _create_teacher(self) -> None:
```

Comment:
I think that we should separate parsing from creating. This method does two tasks now. Maybe we should parse everything first, throw all errors, and once they're all fixed, we create teachers in the database.

Reply:
I think we should validate everything, then populate all. We don't need any data just to check the database, we're already using dummy data for that

Reply:
Okay, I will create a dry mode